qiita_pet/handlers/rest/rest_handler.py

```diff
@@ -0,0 +1,28 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+from qiita_db.study import Study
+from qiita_db.exceptions import QiitaDBUnknownIDError
+from qiita_pet.handlers.util import to_int
+from qiita_pet.handlers.base_handlers import BaseHandler
+
+
+class RESTHandler(BaseHandler):
+    def fail(self, msg, status):
```

Comment:
I'm unsure if you are going to be able to modify the QiitaHTTPError to write the message in the body (is not a handler class so I'm unsure on how tornado is handling that internally). It may be worth testing how the current status of the QiitaHTTPError shows when using curl.

I don't think registering these errors in the DB is needed as these are misuses of the API and generally user generated. That would be an easy way of blowing up the size of our database.